### PR TITLE
[license] Add notice files for binary release

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

For binary release, if a dependency has NOTICE file, we need to copy it to the binary release package.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: dist/licenses/LICENSE.jboss-marshalling.txt [hunk 1]
```diff
@@ -0,0 +1,179 @@
+
+                                 Apache License
+                           Version 2.0, January 2004
+                        https://www.apache.org/licenses/
+
+   TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
+
+   1. Definitions.
+
+      "License" shall mean the terms and conditions for use, reproduction,
```
Comment (CalvinKirs) [inline dist/licenses/LICENSE.jboss-marshalling.txt:10]: With a standard AL2 like this, we don't need to add it.

**Comment by CalvinKirs** [comment]
As far as i know, we have three binary packages, we'd better create the corresponding directory separately to manage the corresponding License and NOTICE,

File: dist/licenses/LICENSE.jboss-marshalling.txt (hunk 1 above)
Comment (morningman) [reply dist/licenses/LICENSE.jboss-marshalling.txt:10]: The file is referenced in NOTICE-dist.txt (Content of NOTICE for hadoop-auth-2.8.0). So I had to add it.

**Review by yangzhg — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
> As far as i know, we have three binary packages, we'd better create the corresponding directory separately to manage the corresponding License and NOTICE,

ok, i see

**Pull request merged**